flaky: TestPrometheus/native_histogram_round-trip_fidelity returns cached series count

`TestPrometheus/native_histogram_round-trip_fidelity` ([integration/prometheus_test.go:390-428](https://github.com/trickstercache/trickster/blob/main/integration/prometheus_test.go#L390-L428)) intermittently fails because its query string is identical to the one used by the earlier subtest `TestPrometheus/native_histogram_instant_query` ([line 303](https://github.com/trickstercache/trickster/blob/main/integration/prometheus_test.go#L303)). OPC caches the first response under the same key; the fidelity subtest then hits cache and compares stale series count against a fresh direct prom query.

CI failure example: https://github.com/trickstercache/trickster/actions/runs/25804239551/job/75802382274

```
prometheus_test.go:410:
    Error:      Not equal: 
                expected: 15
                actual  : 6
    Messages:   Trickster proxy should return same number of series as direct Prometheus query
--- FAIL: TestPrometheus/native_histogram_round-trip_fidelity
```

The cached response was written by the earlier subtest when prom had N series. By the time the fidelity subtest runs, prom has accumulated more (test traffic adds scrape history), so `direct=N+k` vs `cached-proxy=N`.

Fix: make the fidelity subtest's query unique per-run -- the histogram subtests on lines 340 and 380 already use `+ 0*<nonce>` for exactly this reason. Or assert cache-miss before comparing.

--- a/integration/prometheus_test.go
+++ b/integration/prometheus_test.go
@@ -388,7 +388,8 @@ func TestPrometheus(t *testing.T) {
 	})
 
 	t.Run("native histogram round-trip fidelity", func(t *testing.T) {
-		params := url.Values{"query": {"prometheus_http_request_duration_seconds"}}
+		query := fmt.Sprintf("prometheus_http_request_duration_seconds + 0*%d", time.Now().UnixNano())
+		params := url.Values{"query": {query}}
 
 		// Query directly against Prometheus
 		prDirect, _ := queryTricksterProm(t, "127.0.0.1:9090", "", "/api/v1/query", params)

